docs/technical-documentation/migrate-csv.md

```diff
@@ -1 +1,815 @@
+## Table of Contents
+
+* [Summary](#summary)
+* [Introduction](#introduction)
+* [Dependencies](#dependencies)
+* [Overview](#overview)
+* [Installation](#installation)
+* [Ingesting Files](#ingesting-files)
+    * [Anatomy of a migration](#anatomy-of-a-migration)
+        * [Source](#source)
+        * [Process](#process)
+        * [Destination](#destination)
+        * [The Process Section in Depth](#the-process-section-in-depth)
+    * [Running the File Migration](#Running-the-file-migration)
+* [Ingesting Nodes](#ingesting-nodes)
+    * [Complex Fields](#complex-fields)
+    * [Running the node migration](#running-the-node-migration)
+* [Migrating Media](#migrating-media)
+* [What have we learned](#what-have-we-learned)
+* [Where to go from here](#where-to-go-from-here)
+
+
+## Summary
+
+This tutorial introduces you to the Drupal 8 Migrate tools available to create Islandora content. Whether you will eventually use CSVs or other sources (such as XML or directly from a 7.x Islandora) this tutorial should be useful as it covers the basics and mechanics of migration.
+
+This tutorial uses the configurations and code available in the [migrate_islandora_csv](https://github.com/Islandora-CLAW/migrate_islandora_csv) module which, when enabled, will create three example migrations ready for you to use with the Migrate API. Each migration comes from one of the files in the `config/install` folder. We'll walk through them in detail below. The module is also uses [Features](https://www.drupal.org/project/feature) which allows you to make changes to the configuration files and sync those changes into your Drupal site.
+
+!!! note "A note on using Features"
+    This tutorial (and Islandora in general) makes heavy use of Features, which is an easy way to ship and install Drupal configuration. However, after enabling a Feature module, the code in that module's directory is no longer "live", as the configuration now resides in the Drupal database. If you change code in the YAML files, it will not take effect until you re-import the Feature. There is a walkthrough in the "Configuration" section of the [Migrate 7.x to 8](https://github.com/Islandora-Devops/migrate_7x_claw) tutorial. 
+
+Sample CSV and images are also included in the module as a convenience so they are easily available on the Drupal server running the migration. (This is not the recommended method for making files available to Drupal in a real migration.)
+
+The module also contains a Migrate process plugin that transforms strings into associative arrays. This is useful for populating multiple Linked Agent fields. It will be available when this module is enabled, and the node migration uses it. It was written generically and will hopefully become part of Migrate Plus, but for now it is here.
+
+When you are ready to create your actual migrations, the contents of this repository can function as a template for you to create the yml files defining your own migrations.
```

Comment:
Is it YAML files?

Reply:
It is YAML. Should we swap this "yml" with YAML and then add a note that YAML files are also commonly referred to by their extension "yml"?

Reply:
I should note that we make references to YAML as "YAML", "yaml", and "yml" in various locations throughout the documentation site. This sounds like something for the styleguide, @manez.

Reply:
Which one is correct?

Reply:
Well, "YAML" is the formal name, but referring to a format by its extension ("yml" in this case) is also common practice, if perhaps informal. "yaml" is the only one that is technically incorrect.